File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    var answer = 0;
    let tmp = new Set();
    
    function isPrime(num) {
        if(num <= 1)
            return false;
        if(num === 2 || num === 3)
            return true;
        
        for(let i=2; i<=Math.sqrt(num); i++) {
           if(num%i === 0)
               return false;
        }
        
        return true;
    }
    
    function permutation(arr, result) {
        if(arr.length >= 1) {
            for(let i=0; i<arr.length; i++){
                let nResult = result + arr[i];
                let nArr = [...arr];
                nArr.splice(i, 1);
                
                // set 자동으로 중복 제거
                if(isPrime(Number(nResult)))
                    tmp.add(Number(nResult));
                
                permutation(nArr, nResult);
            }
        }
    }
    
    permutation(numbers, "");
    
    answer = tmp.size;
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <set>
#include <algorithm>

using namespace std;

bool isPrime(int n){
    if(n == 0 || n == 1)
        return false;
    if(n == 2 || n == 3)
        return true;
    
	for (int i=2; i<=sqrt(n); i++) {
        if (n%i == 0)
			return false;
	}
    
	return true;
}

int solution(string numbers){
    int answer = 0;
    set<int> num_s;
    
    sort(numbers.begin(), numbers.end());
    
    do {
        string tmp = "";
        for(int i=0; i<numbers.size(); i++){
            tmp += numbers[i];
            if(isPrime(stoi(tmp)))
                num_s.insert(stoi(tmp));
        }
    } while (next_permutation(numbers.begin(), numbers.end()));
    
    answer = num_s.size();
    
    return answer;
}